### DISCUSSION TITLE
How to include full file path in junit assertion failure format?

### DISCUSSION THREAD

**Discussion by mohkale** [junit-team/junit-framework#4037] [Q&A]
Hi,

I'd like to integrate junit into my Emacs development workflow. Emacs provides the functionality to run shell commands and then parse the output of them to populate a jump list and then quickly jump to each of those failures. It relies on the compilation/lint tools encoding the location info of those failures to make it simple to jump to.

For example if I break a java file and run a build with `./gradlew --daemon --parallel test` then I'll get printed:

```text
/tmp/tmp.t36cfaG8Yv/list/src/test/java/org/example/list/LinkedListTest.java:13: error: ';' expected
        assertEquals(1, list.size())
```

The `path:line-number: error: message` is a format Emacs can recognize so this error is jumpable. Unfortunately the output format of junit is just the java file name with no details about where the file is located so if you just run all tests from the root of the project you've gotta figure out where the directory containing this file is yourself :disappointed:.

```text
LinkedListTest > testConstructor() FAILED
    org.opentest4j.AssertionFailedError at LinkedListTest.java:13
```

Is there a way to encode the full `/tmp/tmp.t36cfaG8Yv/list/src/test/java/org/example/list/` prefix before LinkedTest.java. Preferably even just the path from the project root or where I started the test command (`./gradlew --daemon --parallel test`) would be sufficient.

**Comment by marcphilipp** [comment]
As far as I'm aware, the absolute or project root relative path is not available from the stack trace which is printed by Gradle in this case.

**Comment by sbrannen** [comment]
> Is there a way to encode the full `/tmp/tmp.t36cfaG8Yv/list/src/test/java/org/example/list/` prefix before LinkedTest.java.

I think you'll need to implement a custom `TestExecutionListener`, override the `executionFinished(...)` method, and use `TestExecutionResult.getThrowable()` to access the exception when the status is `TestExecutionResult.Status.FAILED`. From the stack track of the exception, you can craft your own console output using the format you require.
